js/dydx.js

```diff
@@ -0,0 +1,469 @@
+                    '403': AuthenticationError,
+                    '400': ExchangeError,
+                },
+                'broad': {
+                    'See /corsdemo for more info': AuthenticationError,
+                    'Invalid signature for onboarding request': AuthenticationError,
+                    'Invalid signature for ApiKey request': AuthenticationError,
+                    'Not Found': ExchangeNotAvailable,
+                    'market must be a valid market (BTC-USD, etc)': BadSymbol,
+                    'Invalid value': ExchangeError,
+                    'AccountNotFoundError': AuthenticationError,
+                    'market must be a valid market': BadSymbol,
+                    'Invalid market': BadSymbol,
+                    'Order is below minimum size of 20000000000000000000': InvalidOrder,
+                    'Order would put account under the collateralization threshold': InsufficientFunds,
+                    'insufficient funds for gas * price + value': InsufficientFunds,
+                },
+            },
+            'requiredCredentials': {
+                'ethereumAddress': true,
+                'privateKey': true, // Ethereum Key Authentication
+                'apiKey': true, // API Key Authentication
+                'secret': true, // API Key Authentication
+                'passPhrase': true, // API Key Authentication
+                // 'starkKeyYCoordinate': true, // STARK Key Authentication
+                // 'starkKey': true, // STARK Key Authentication
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const configResponse = await this.publicGetConfig ();
+        //
+        // {
+        //     collateralAssetId: '0x02893294412a4c8f915f75892b395ebbf6859ec246ec365c3b1f56f47c3a0a5d',
+        //     collateralTokenAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+        //     defaultMakerFee: '0.0005',
+        //     defaultTakerFee: '0.001',
+        //     exchangeAddress: '0xD54f502e184B6B739d7D27a6410a67dc462D69c8',
+        //     maxExpectedBatchLengthMinutes: '720',
+        //     maxFastWithdrawalAmount: '200000',
+        //     cancelOrderRateLimiting: {
+        //       maxPointsMulti: '3',
+        //       maxPointsSingle: '8500',
+        //       windowSecMulti: '10',
+        //       windowSecSingle: '10'
+        //     },
+        //     placeOrderRateLimiting: {
+        //       maxPoints: '1000',
+        //       windowSec: '10',
+        //       targetNotional: '40000',
+        //       minLimitConsumption: '4',
+        //       minMarketConsumption: '20',
+        //       minTriggerableConsumption: '100',
+        //       maxOrderConsumption: '100'
+        //     }
+        // }
+        //
+        this['fees']['trading']['maker'] = this.safeString (configResponse['defaultMakerFee']);
+        this['fees']['trading']['taker'] = this.safeString (configResponse['defaultTakerFee']);
```

Comment:
We should not override these values. Instead, we should implement `fetchTradingFees` for this exchange, i guess.